### DISCUSSION TITLE
ReportWidgets - which is correct way (more Winterish :D

### DISCUSSION THREAD

**Discussion by SledgehammerPL** [wintercms/winter#805] [Q&A] [answered]
```php
class MembersAtRisk extends ReportWidgetBase
{
  public function render()
  {
    try {
      $this->loadData();
    }
    catch (Exception $ex) {
      $this->vars['error'] = $ex->getMessage();
    }
    return $this->makePartial('widget');
  }

   protected function loadData()
    {

      $this->vars['members_list'] = # what put here?
    }
}
```
And in members_list I want special filtered list of members (not used anywhere else)

And now there's three ways I can imagine:

1. Make a separate model for this (the danger: it could be MANY models - so I don't like it)
2.  Somehow access to scope (and create scope in Members model) Members::scopeMyScope() doesn't work ;D
3.  Some special function addes to Members model....

**Comment by WebVPF** [comment, marked as answer]
More or less like this

```php
use WebVPF\PlugName\Models\Member;

class MembersAtRisk extends ReportWidgetBase
{
    // ...

    protected function loadData()
    {

      $this->vars['members_list'] = Member::where('is_activated', true)->get();
    }
}
```

Here is an example with data from the Winter.User plugin: https://github.com/wintercms/wn-user-plugin/blob/7208cc1af5b8d5af6089eb4c08225c84547a0efd/reportwidgets/Statistics.php

**Reply by SledgehammerPL** [reply]
Simplicity blows my mind....

**Comment by SledgehammerPL** [comment]
But for complex views you can use: 

$this->vars['members_list'] = Db::select('select * from users where active = ?', [1]);

[https://wintercms.com/docs/database/basics](url)

**Reply by SledgehammerPL** [reply]
I don't like this solution but for now I can't see any better - because I have very complex views which are nested with complex views... I think that creation of view during every time when user shows page makes process uneffective... I
